src/AccessibilityInsights.Win32/Win32.csproj

```diff
@@ -14,12 +14,22 @@
     <BuildNumber>1.1.$([System.Math]::Round($([System.DateTime]::ParseExact($(CurrentDateString),`yyyy-MM-dd`,null).Subtract($(CountDate)).TotalDays)).ToString(`0000`))$(Revision)</BuildNumber>
   </PropertyGroup>
 
-  <Target Name="CreateVersionFile">
-    <WriteLinesToFile File="$(MSBuildProjectDirectory)\..\bld\A11yInsightsVersionInfo.cs" Overwrite="True" Lines="[assembly: System.Reflection.AssemblyVersion(&quot;$(BuildNumber)&quot;)] // Generated by build" />
+  <Target Name="CreateVersionFile" BeforeTargets="BeforeCompile">
+    <WriteLinesToFile File="$(MSBuildProjectDirectory)\..\bld\A11yInsightsVersionInfo.cs"
```

Comment:
Instead of this strategy, did you consider just adding `WriteOnlyWhenDifferent="true"` to the `WriteLinesToFile` directive? I think that would:

* still avoid unnecessary rebuilds (the version file timestamp won't be updated unless necessary, so it won't trigger a rebuild unless actually changed)
* be simpler (wouldn't need the separate clean step)
* be more correct if someone leaves Visual Studio running on a dev box overnight (rebuilds with the new date's version number the next morning even without an explicit clean)

Reply:
I love this idea... but it didn't work. I modified the .csproj file to generate a different version, and it never overwrote the file with the new content. The only option that I've been able to reliable make work is to have the clean target adn set `Overwrite` to true.

Reply:
OK, finally got it to work. I had forgotten to remove the condition to check for the file existence 🤦